File: guests/java/dev/kaya/milestone2kt/Confirm.java
package dev.kaya.milestone2kt;

import dev.kaya.KayaApp;
import dev.kaya.KayaWire;

/**
 * The confirm conformance scene from the JVM — the modal-alert grammar
 * via the chain spelling. The three rounds take the three answer paths
 * (action 0, action 1, and KayaWire.ALERT_CHOICE_CANCEL, which is every
 * platform-native dismissal). See guests/rust/confirm.rs and
 * tools/scenes/confirm.steps.
 */
final class Confirm {
    static void app() {
        KayaApp app = new KayaApp();

        app.build(tx -> {
            tx.window(0).title("confirm");
            KayaApp.Signal<String> status = tx.signal("no decision");
            tx.mount(tx.column(() -> {
                tx.label(status); // label#0
                tx.button("delete", inner -> {
                    inner.showAlert()
                            .title("delete item?")
                            .message("this cannot be undone")
                            .action("Delete")
                            .action("Archive")
                            .cancel("Keep")
                            .onResult((tx2, choice) -> {
                                if (choice == KayaWire.ALERT_CHOICE_CANCEL) {
                                    tx2.write(status, "kept");
                                } else if (choice == 1) {
                                    tx2.write(status, "archived");
                                } else {
                                    tx2.write(status, "deleted");
                                }
                            })
                            .show();
                });
                tx.button("eject", inner -> {
                    inner.showAlert()
                            .title("eject disk?")
                            .message("it is still mounted")
                            .action("Eject")
                            .cancel("Hold")
                            .onResult((tx2, choice) -> {
                                tx2.write(status,
                                        choice == KayaWire.ALERT_CHOICE_CANCEL
                                                ? "held" : "ejected");
                            })
                            .show();
                });
            }));
            return null;
        });

        app.dispatchLoop();
    }

    private Confirm() {}
}

File: guests/go/confirm/confirm.go
// The confirm conformance scene, Go port — the modal-alert grammar
// via the chain spelling. Three rounds take the three answer paths
// (action 0, action 1, kaya.AlertChoiceCancel). See
// guests/rust/confirm.rs and tools/scenes/confirm.steps.
package confirm

import (
	kaya "dev.kaya/bindings/go"
)

func App() *kaya.App {
	app := kaya.NewApp()

	var status kaya.Signal[string]
	app.Build(func(tx *kaya.Tx) {
		tx.Window(0).Title("confirm")
		status = tx.Signal("no decision")

		tx.Mount(tx.Column(func() {
			tx.Label(status) // label#0
			tx.Button("delete", func(tx *kaya.Tx) {
				tx.ShowAlert().
					Title("delete item?").
					Message("this cannot be undone").
					Action("Delete").
					Action("Archive").
					Cancel("Keep").
					OnResult(func(tx *kaya.Tx, choice uint32) {
						switch choice {
						case 0:
							tx.Write(status, "deleted")
						case 1:
							tx.Write(status, "archived")
						case kaya.AlertChoiceCancel:
							tx.Write(status, "kept")
						}
					}).
					Show()
			})
			tx.Button("eject", func(tx *kaya.Tx) {
				tx.ShowAlert().
					Title("eject disk?").
					Message("it is still mounted").
					Action("Eject").
					Cancel("Hold").
					OnResult(func(tx *kaya.Tx, choice uint32) {
						if choice == kaya.AlertChoiceCancel {
							tx.Write(status, "held")
						} else {
							tx.Write(status, "ejected")
						}
					}).
					Show()
			})
		}))
	})

	return app
}
